src/OrbitGl/TimerTrack.cpp

```diff
@@ -186,48 +155,82 @@ void TimerTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick,
           // depth are drawn below each other (and thus do not overlap).
           if (timer_info.end() > next_timer_info.start() &&
               next_timer_info.type() == timer_info.type()) {
-            double intersection_end_us = end_us;
-            end_us = time_graph_->GetUsFromTick(next_timer_info.start());
-
-            WorldXInfo world_x_info = ToWorldX(end_us, intersection_end_us, inv_time_window,
-                                               world_start_x, world_width, canvas->GetWidth());
-            if (world_x_info.is_visible_width && !is_collapsed) {
-              Vec3 bottom_left(world_x_info.world_x_start, world_timer_y, z);
-              Vec3 bottom_right(world_x_info.world_x_start + world_x_info.world_x_width,
-                                world_timer_y, z);
-              Vec3 top_left(world_x_info.world_x_start, world_timer_y + box_height, z);
-              Triangle triangle(bottom_left, bottom_right, top_left);
-
-              auto user_data = std::make_unique<PickingUserData>(
-                  &text_box, [&](PickingId id) { return this->GetBoxTooltip(id); });
-              batcher->AddTriangle(triangle, color, std::move(user_data));
-            }
+            end_or_next_start_us = time_graph_->GetUsFromTick(next_timer_info.start());
           }
         }
 
-        // If the timer is overlapping completely with the previous and next timer, we neither draw
-        // text nor a box.
-        if (start_us > end_us) continue;
-
         double elapsed_us = end_us - start_us;
-        WorldXInfo world_x_info = ToWorldX(start_us, end_us, inv_time_window, world_start_x,
-                                           world_width, canvas->GetWidth());
 
-        Vec2 pos(world_x_info.world_x_start, world_timer_y);
-        Vec2 size(world_x_info.world_x_width, GetTextBoxHeight(timer_info));
-        text_box.SetPos(pos);
-        text_box.SetSize(size);
+        // Draw the Timer's text if it is not collapsed.
+        if (!is_collapsed) {
+          // For overlaps, let us make the textbox just a bit wider:
+          double left_overlap_width_us = start_or_prev_end_us - start_us;
+          double text_x_start_us = start_or_prev_end_us - (.25 * left_overlap_width_us);
```

Comment:
As I told you, I don't like this. There are cases that it won't work and I don't like that. I will think some easier way to do it more correctly.

Reply:
BTW, if you want to make it anyways, you should create a constant.

Reply:
I assume this is talking about the .25? Why do we need to make the textbox a bit wider? Is there an alternative?

Reply:
Basically this is the issue:
<img width="569" alt="name" src="https://user-images.githubusercontent.com/8610429/106602549-725d1980-655d-11eb-94b4-7d599a94dcce.png">

Still thinking this approach works pretty well and does not require "complex" computations, where I would need to calculate the intersection the character height and the triangle.

Reply:
Ok, thanks for the screenshot. So, things look a bit funky when you have a long overlap. 

Yes, it seems you would have to know the character height to align it well to the slanted edge as good as possible. 

I don't feel strongly either way. I would say to go with the simple solution and if we see issues and it bothers us a lot, we'll look into it again later.

Reply:
Just to clarify, Pablo provided the screenshot. That was from the version prior the 25% change. After that, you basically have the text perfectly extended in the given case. However, it is just an approximation and there might be situations where this is not pixel perfect.

Reply:
After thinking a little bit more, I regret about my previous comment. I'm totally happy with this solution. Maybe we could modify the left margin to be 10% or 15% for the underscore (you could check the first character of the text, but that could be tooo much). But I agree with FlorianK here.